[12.0][MIG] Migration of stock_picking_auto_create_lot

meigallodixital opened the pull request:
Migration proposal

sergio-teruel committed:
[11.0][ADD] stock_picking_auto_create_lot: Create lot for incoming pickings

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update stock_picking_auto_create_lot.pot

sergio-teruel committed:
[11.0][FIX] stock_picking_auto_create_lot: Not create lot if lot_name is stored in move_line

meigallodixital committed:
[MIG] stock_picking_auto_create_lot: Migration to 12.0

pedrobaeza commented:
Please sign the CLA

pedrobaeza commented on stock_picking_auto_create_lot/i18n/es.po:
```diff
@@ -1,44 +1,42 @@
+#: model:ir.model.fields,field_description:stock_picking_auto_create_lot.field_stock_picking_type__auto_create_lot
 msgid "Auto Create Lot"
-msgstr "Auto crear lote"
+msgstr "Auto Crear Lote"
```
You can respect previous translation, or even better, try to unify the source term (with or without upper case in the middle) for not having to translate 2 times the same.

meigallodixital replied:
> Please sign the CLA

I signed and send icla yesterday

pedrobaeza replied:
OK, then our OCA secretary will insert it and with next commit the acknowledge will appear :)

pedrobaeza commented on stock_picking_auto_create_lot/static/description/index.html:
```diff
@@ -389,7 +389,7 @@ <h1><a class="toc-backref" href="#id1">Configuration</a></h1>
 <ol class="arabic simple">
 <li>Go to a <em>Inventory &gt; Settings &gt; Operation Types</em>.</li>
-<li>Set ???auto create lot??? option for this operation type.</li>
+<li>Set ‘auto create lot’ option for this operation type.</li>
```
Changing this is futile, as it's overwritten again by the bot, as it's auto-generated from pieces in `readme` folder. You can even pre-generate it using https://github.com/OCA/maintainer-tools/#readme-generator

meigallodixital replied:
I try readme generator, thanks

meigallodixital replied:
I have regenerated the readme with the generator but I have two doubts:

- Should the changelog be changed in the first version of a migration? Only the changes of the 12.0 changes or the previous changes (11.0) are kept too?
- I see that there is a badge 5 with a runbot in the index.html, but I do not see where it comes from. Does that part have to be edited by hand or does the bot modify it?

pedrobaeza replied:
That badge is autogenerated by the README script (run manually with proper arguments or by the bot when this is merged), and it links to the proper branch and repo.

About the changelog, can be considered since the beginning, but personally I'm against that changelog because sometimes there's going to be things that are fixed simultaneously in several branches with a lot of mess in that history. I never put it in my modules (it's optional).

meigallodixital committed:
README generated with OCA README generator

pedrobaeza commented on stock_picking_auto_create_lot/README.rst:
```diff
@@ -14,13 +14,13 @@ Stock Picking Auto Create Lot
     :alt: License: AGPL-3
 .. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fstock--logistics--workflow-lightgray.png?logo=github
-    :target: https://github.com/OCA/stock-logistics-workflow/tree/11.0/stock_picking_auto_create_lot
+    :target: https://github.com/OCA/stock-logistics-workflow/tree/10.0/stock_picking_auto_create_lot
```
It seems you have passed incorrect version information to the command line...

meigallodixital replied:
fixed, sorry fot the typo :(

meigallodixital committed:
FIX - Incorrect branch on README.rst and index.html

tbaden commented on stock_picking_auto_create_lot/models/product.py:
```diff
@@ -0,0 +1,11 @@
+    _inherit = 'product.template'
+
+    auto_create_lot = fields.Boolean(
+        string='Auto create lot',
```
please use the same string, either with or without uppercase. (see below)

tbaden commented on stock_picking_auto_create_lot/models/stock_picking.py:
```diff
@@ -0,0 +1,24 @@
+class StockPickingType(models.Model):
+    _inherit = 'stock.picking.type'
+
+    auto_create_lot = fields.Boolean(string='Auto Create Lot')
```
please use the same string, either with or without uppercase. (see above)

tbaden reviewed (commented):
as mentioned from pedro please unify the string (see below), this way you only need to translate one term and can delete the other one.

everything else looks good to me

HaraldPanten reviewed (approved):
Functional review. Tested on runbot 👍

HaraldPanten commented:
Hi @meigallodixital ,

How's this PR? It's functionally working fine.

Thanks!

rousseldenis commented:
@meigallodixital

sergio-teruel commented on stock_picking_auto_create_lot/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    'name': 'Stock Picking Auto Create Lot',
+    'summary': 'Auto create lots for incoming pickings',
+    'version': '12.0.1.0.0',
+    'development_status': 'Beta',
```
You can change by "Production/Stable"

sergio-teruel commented on stock_picking_auto_create_lot/i18n/es.po:
```diff
@@ -0,0 +1,42 @@
+#. module: stock_picking_auto_create_lot
+#: model:ir.model,name:stock_picking_auto_create_lot.model_stock_picking
+msgid "Transfer"
+msgstr "Transferir"
```
Add missing line

sergio-teruel reviewed (approved):
two comments only

pedrobaeza committed:
[IMP] stock_picking_auto_create_lot: Change latest comments:

- Promoted to Production/Stable
- Put maintainer
- Regenerate README
- Unify string
- Adapt translations accordingly

pedrobaeza commented:
I have pushed an extra commit for fixing pending comments, so we can

/ocabot merge
